clean_project/index.py

```diff
@@ -0,0 +1,87 @@
+"""
+Create a JSON file used to indexing all words in a config_file.yaml
+"""
+import argparse
+import codecs
+import clean_file
+import psycopg2
+import yaml
+
+from collections import defaultdict
+from ConfigParser import SafeConfigParser
+
+
+def load_config_file(connect_path):
+    config = SafeConfigParser()
+    config.read(connect_path)
+    list_keys = {'database': config.get('db_connection', 'database'),
+                 'user': config.get('db_connection', 'user'),
+                 'password': config.get('db_connection', 'password')}
+    return list_keys
+
+
+def index(config_path):
+    """
+    Build a dictionary, given a configuration files
+    :param config_path: Configuration path which contains a list of files that will be cleaned
+    :return: Index object
+    """
+    with open(config_path) as config_file:
+        data = yaml.load(config_file)
+    if data is not None:
+        index_words = defaultdict(lambda: defaultdict(list))
+        for file_dirty in data['files_path']:
+            with codecs.open(file_dirty.get('file'), 'r', 'utf-8') as source_file:
+                # Used 1-based indexing for showing line numbers in output representation
+                for i, line in enumerate(source_file, 1):
+                    cleaned_line = clean_file.sanitize(line)
+                    for word in cleaned_line.split():
+                        index_words[word.lower()][source_file.name].append(i)
+    else:
+        raise ValueError('Empty configuration file')
+    return index_words
+
+
+def build_index(index_words, connect_path):
+    """
+    Insert the index in a specific database for store all data
+    :param index_words: Object which contains the built index
+    """
+    connection = None
+    cursor = None
+    db_keys = load_config_file(connect_path)
+    try:
+        connection = psycopg2.connect(database=db_keys.get('database'),
+                                      user=db_keys.get('user'),
+                                      password=db_keys.get('password'))
+        cursor = connection.cursor()
+        # Update index table only if word and file match. Else do nothing
+        query_update = "UPDATE index SET occurrences=%s WHERE word=%s AND file=%s;"
+        # Insert only if word and file don't exist in index table. Else do nothing
+        query_insert = ("INSERT INTO index (word, file, occurrences)"
+                        "SELECT %s, %s, %s"
+                        "WHERE NOT EXISTS (SELECT 1 FROM index WHERE word=%s AND file=%s);")
```

Comment:
Both query_update and query_insert are constants that can go on top of the file.